src/subdomains/generic/user/models/user-data/user-data.service.ts

```diff
@@ -220,6 +223,16 @@ export class UserDataService {
     await this.userDataRepo.update(...userData.blockUserData());
   }
 
+  async nameCheck(userData: UserData): Promise<boolean> {
+    if (Util.daysDiff(userData.lastCheckedTimestamp) <= 1) return true;
+
+    const nameCheck = await this.nameCheckLogService.doNameCheck(userData);
+    if (nameCheck === RiskRate.SANCTIONED) return false;
+
+    await this.userDataRepo.update(...userData.refreshLastCheckedTimestamp());
+    return true;
+  }
```

Comment:
Maybe we can find a meaningful name here of what it returns? Maybe something like `isSanctioned`?